packages/@aws-cdk/aws-kms/lib/key.ts

```diff
@@ -675,3 +675,20 @@ export class Key extends KeyBase {
     }));
   }
 }
+
+/**
+ * Whether the given object is a Construct
+ *
+ * Normally we'd do `x instanceof Construct`, but that is not robust against
+ * multiple copies of the `constructs` library on disk. This can happen
+ * when upgrading and downgrading between v2 and v1, and in the use of CDK
+ * Pipelines is going to an error that says "Can't use Pipeline/Pipeline/Role in
+ * a cross-environment fahsion", which is very confusing.
+ */
+function isConstruct(x: any): x is Construct {
+  const sym = Symbol.for('constructs.Construct.node');
+  return (typeof x === 'object' && x &&
+    (x instanceof Construct // happy fast case
+    || !!(x as Construct).node // constructs v10
+    || !!(x as any)[sym])); // constructs v3
+}
```

Comment:
This is how production software looks like!

Reply:
Should we move this to the `constructs` library? Feels like it might be a better place to put this code, no?

Reply:
For sure. Don't forget to add it to both versions :)